### DISCUSSION TITLE
Zoraxy update to 3.1.2 failed

### DISCUSSION THREAD

**Discussion by Hindin81** [community-scripts/ProxmoxVE#25] [Q&A] [answered]
Hello when you want to update Zoraxy to 3.1.2 then the following error message appears:

 \rm: cannot remove '/opt/zoraxy/zoraxy': No such file or directory

[ERROR] in line 63: exit code 0: while executing command rm /opt/zoraxy/zoraxy

kind regards
Daniel

**Comment by havardthom** [comment]
From changelog:
## 2024-10-25

### Changed

- **Zoraxy LXC** [(Commit)](https://github.com/tteck/Proxmox/commit/468a5d367ded4cf453a1507452e112ac3e234e2a)
  - Switch built from source to a pre-compiled binary version.
  - Breaking Change

**Reply by JcMinarro** [reply]
How migration to the new pre-compiled binary should be done?

**Reply by dsiebel** [reply]
Hey, there! 
I was the one who did the [original implementation](https://github.com/tteck/Proxmox/pull/3970), here's what I did to migrate my existing Zoraxy LXCs:

```
systemctl stop zoraxy.service
mv /opt/zoraxy/src /opt/data/
rm -r /opt/data/Makefile /opt/data/*.go /opt/data/go.mod /opt/zoraxy/go.sum /opt/zoraxy/start.sh /opt/zoraxy/
mv /opt/data /opt/zoraxy
sed -i 's/ExecStart=\/opt\/zoraxy\/src\/.\/zoraxy/ExecStart=\/opt\/zoraxy\/.\/zoraxy/g' /etc/systemd/system/zoraxy.service
sed -i 's/WorkingDirectory=\/opt\/zoraxy\/src\//WorkingDirectory=\/opt\/zoraxy\//g' /etc/systemd/system/zoraxy.service
systemctl daemon-reload
update
```

EDIT: adopting suggestions from @JcMinarro [below](https://github.com/community-scripts/ProxmoxVE/discussions/25#discussioncomment-11136224)

**Reply by JcMinarro** [reply]
The [update script](https://github.com/community-scripts/ProxmoxVE/blob/main/ct/zoraxy.sh#L63) will keep failing because the isn't any `/opt/zoraxy/zoraxy` files to be removed.

I will try to update later, but an extra step would be needed, at least touching the file `touch /opt/zoraxy/zoraxy` before `update` is run

**Reply by JcMinarro** [reply]
:warning: the directory `/opt/zoraxy/src` contains the user config, do not remove it before you copy its data to a new folder

**Comment by dsiebel** [comment]
> Hello when you want to update Zoraxy to 3.1.2 then the following error message appears:
> 
> \rm: cannot remove '/opt/zoraxy/zoraxy': No such file or directory
> 
> [ERROR] in line 63: exit code 0: while executing command rm /opt/zoraxy/zoraxy
> 
> kind regards Daniel

I just tried it on one of my Zoraxy LXC and it worked just fine

```
 _____
/__  /  ____  _________ __  ____  __
  / /  / __ \/ ___/ __ `/ |/_/ / / /
 / /__/ /_/ / /  / /_/ />  </ /_/ /
/____/\____/_/   \__,_/_/|_|\__, /
                           /____/
 ✓ Updated Zoraxy
```

What did you run to update it?

**Reply by Hindin81** [reply]
When I use: touch /opt/zoraxy/zoraxy and then the command update, it says updated. But Zoraxy is still in Version 3.1.1 active and not the new one in 3.1.2.

**Comment by havardthom** [comment]
Edit: see https://github.com/community-scripts/ProxmoxVE/discussions/25#discussioncomment-11136224

**Reply by JcMinarro** [reply]
:warning: the directory `/opt/zoraxy/src` contains the user config, do not remove it before you copy its data to a new folder

**Comment by JcMinarro** [comment, marked as answer]
@dsiebel @havardthom Please, update the commands you are using in your comments because you are removing the directory `/opt/zoraxy/` but it contains any previous data the user had.

Those are the steps I have followed to update Zoraxy LXC:

```
systemctl stop zoraxy.service
mv /opt/zoraxy/src /opt/data/
rm -r /opt/data/Makefile /opt/data/*.go /opt/data/go.mod /opt/zoraxy/go.sum /opt/zoraxy/start.sh /opt/zoraxy/
mv /opt/data /opt/zoraxy
sed -i 's/ExecStart=\/opt\/zoraxy\/src\/.\/zoraxy/ExecStart=\/opt\/zoraxy\/.\/zoraxy/g' /etc/systemd/system/zoraxy.service
sed -i 's/WorkingDirectory=\/opt\/zoraxy\/src\//WorkingDirectory=\/opt\/zoraxy\//g' /etc/systemd/system/zoraxy.service
systemctl daemon-reload
update
```

**Reply by dsiebel** [reply]
updated my comment.
